pynxtools_stm/stm_file_parser.py

```diff
@@ -283,8 +269,15 @@ def fill_out_NXdata_group():
                         template[signal_attr] = data_field_nm.lower()
                     else:
                         template[auxiliary_signals_attr].append(data_field_nm.lower())
-                for axis, axis_data in zip(axes_name, axes_data):
+
+                if len(axes_data) != len(axes_units):
+                    missing = len(axes_data) - len(axes_units)
+                    axes_units.extend([""] * missing)
+                for axis, axis_data, unit in zip(axes_name, axes_data, axes_units):
+                    template[f"{nxdata_grp}/{axis}"] = axis_data
```

Comment:
Lines 277 and 278 are the same, but you are not using `axes_name`. I guess that's unintended.

Reply:
Noted!